### PR TITLE
fix: add traces to updates with generared keys

### PR DESCRIPTION
**Pull request by ajkaanbal**
I was having trouble tracing `updates` using `withUniqueGeneratedKeys`, these changes appear to have solved the problem so far.

### CODE REVIEW

File: natchez-extras-doobie/src/main/scala/com/ovoenergy/natchez/extras/doobie/TracedStatement.scala [hunk 1]
```diff
@@ -14,7 +14,8 @@ import scala.annotation.nowarn
 @nowarn
 private[doobie] case class TracedStatement(
   p: PreparedStatement,
-  queryString: String
+  queryString: String,
+  c: Any*
```
Comment (voidcontext) [inline natchez-extras-doobie/src/main/scala/com/ovoenergy/natchez/extras/doobie/TracedStatement.scala:18]: What does `c` stand for? Could you please use a more descriptive name?

I see that we didn't set a good example with `p` for `preparedStatement` on line 16, but we should do better in the future.

File: natchez-extras-doobie/src/main/scala/com/ovoenergy/natchez/extras/doobie/TracedTransactor.scala [hunk 2]
```diff
@@ -93,12 +93,24 @@ object TracedTransactor {
...
+          override def prepareStatement( a: String, b: Array[String]): Kleisli[F, Connection, PreparedStatement] =
+            super.prepareStatement(a, b).map(TracedStatement(_, a, b): PreparedStatement)
+          override def prepareStatement(a: String, b: Array[Int]): Kleisli[F, Connection, PreparedStatement] =
+            super.prepareStatement(a, b).map(TracedStatement(_, a, b): PreparedStatement)
+          override def prepareStatement(a: String, b: Int) =
+            super.prepareStatement(a, b).map(TracedStatement(_, a, b): PreparedStatement)
+          override def prepareStatement(a: String, b: Int, c: Int) =
+            super.prepareStatement(a, b, c).map(TracedStatement(_, a, b, c): PreparedStatement)
+          override def prepareStatement(a: String, b: Int, c: Int, d: Int) =
+            super.prepareStatement(a, b, c, d).map(TracedStatement(_, a, b, c, d): PreparedStatement)
```
Comment (voidcontext) [inline natchez-extras-doobie/src/main/scala/com/ovoenergy/natchez/extras/doobie/TracedTransactor.scala:112]: It seems the variadic parameter is not used anywhere, why is it needed to be set?

**Comment by voidcontext** [comment]
@ajkaanbal Could you please describe the issue / symptom you were experiencing?

**Comment by richard-dennehy-ovo** [comment]
Hi @voidcontext, I believe we ran into the same issue; queries using `withUniqueGeneratedKeys` don't appear to get traced at all, e.g. https://github.com/ovotech/carbon-zero-ingestor/blob/b58cc5ac51d2aab28d69a284b34f09d99ecd9877/service/src/main/scala/com/ovoenergy/carbonzero/ds/billbreakdown/weekly/BillBreakdownWeeklyStorage.scala#L97

**Comment by voidcontext** [comment]
HI @ajkaanbal and @richard-dennehy-ovo, I had some time today so I added the suggested changes in #114 and merged it, I'll tag a fix release soon, it should be available in maven later today.

**Pull request merged**